src/lua.cpp

```diff
@@ -242,6 +245,80 @@ static int disassembleTeakThunk(lua_State* L) {
 	return 1;
 }
 
+#ifndef __ANDROID__
+// Haptics functions
+
+namespace Haptics {
+	std::unique_ptr<ExternalHapticsManager> hapticsManager = nullptr;
+
+	static int initHapticsThunk(lua_State* L) {
+		if (hapticsManager == nullptr) {
+			hapticsManager.reset(new ExternalHapticsManager());
+		}
+
+		return 0;
+	}
+
+#define HAPTICS_THUNK(func)                \
```

Comment:
X macros are bad because they enable you to essentially have a very easy and portable way to generate boilerplate code and reduce duplicated code, thus leading to less source code cluttering. Wait why did I say they are bad?

Reply:
lines of code = performance